src/egon/data/processing/osmtgmod/__init__.py

```diff
@@ -74,7 +74,7 @@ def import_osm_data():
         target_path = osm_config["target"]["path_testmode"]
 
     filtered_osm_pbf_path_to_file = os.path.join(
-        egon.data.__path__[0] + "/importing" + "/openstreetmap/"
+        egon.data.__path__[0] + "/datasets" + "/osm/"
```

Comment:
While this certainly fixes the immediate issue of the failing task, there is deeper issues here which does not get addressed. The task downloading the OSM data file puts the downloaded file into the directory of the module where the task's callable is defined instead of the current working directory. A more comprehensive fix would be to:
  - modify to task downloading the OSM data to place the resulting file into (a subdirectory of) the current working directory,
  - locate all places which also access the file and
  - modify these places to read the file from the new location.
This would also make sure there are no other tasks making a similar error. Because those tasks might not even raise an error but silently do the wrong thing.

Also, there is no reason to use `os.path.join` if it's not used with more than one argument. The code above (and below) is strictly equivalent to
```python
filtered_osm_pbf_path_to_file = (
    egon.data.__path__[0] + "/datasets" + "/osm/" + target_path
)
```

The whole point of `os.path.join` is to join paths in a platform independent manner, without having to specify the path separator, `"/"` in the case at hand, manually. So the correct use of `os.path.join` would be
```python
filtered_osm_pbf_path_to_file = os.path.join(
    egon.data.__path__[0], "datasets", "osm", target_path
)
```
This still repeats an error from the old code though, namely using a mixture of generating the path from an imported module and then modifying it to get the path of a different module. The correct way of getting the path of a module is using the module directly
```python
import egon.data.datasets.osm


filtered_osm_pbf_path_to_file = os.path.join(
    egon.data.datasets.osm.__path__[0], target_path
)
```
Doing this would have prevented the error from being so hard to find because importing the `egon.data.importing.openstreetmap` module would have failed after the migration of the OSM dataset.

Last but not least, using the `__path__` attribute is a bit fragile and we moved to using the more robust `importlib_resources` module, whose `files` function does what the `__path__` attribute does, just better, because it fails with an appropriate error message if no path can be determined:

```python
import importlib_resources as resources

import egon.data.datasets.osm


filtered_osm_pbf_path_to_file = os.path.join(
    resources.files(egon.data.datasets.osm), target_path
)
```

Note that `importlib_resources.files` returns a `Path` object, which can be `join`ed to a string (or another `Path`) using the `/` operator. And you also don't have to use the fully qualified module name:
```python
import importlib_resources as resources

from egon.data.datasets import osm


filtered_osm_pbf_path_to_file = str(resources.files(osm) / target_path)
```

Sorry for dumping all of this onto you, @ClaraBuettner, although you only found the error and the appropriate fix and have the least to do with how those errors came about. When reading the fix, I just didn't know of any better place than here, to put the information. But it's intended for @nesnoj and other users of packaged data files, generated data files, `os.path.join` and the `__path__` attribute as well.

Reply:
I didn't do a code review but only tested if it works - shame on me.
@gnn Your comment points to a crucial aspect - in several tasks files are flying around in the task dir(s) but not in the working dir. A correct handling should be part of some best practice docs.
I think we won't fix this now but should definitely approach as soon as all tasks will have been migrated to datasets. I've opened an issue: #341

Reply:
Cool. Now I know that notifications get sent out for comments on merged PRs, too. :)

> I didn't do a code review but only tested if it works - shame on me.

No shaming intended. :) Quite the contrary. Reviews don't always have to be so elaborate because this is pretty time consuming and acts as a blocker most of the time. Definitely more often than not. Checking that new code works is actually the most important thing and essential to get things done. So thanks a lot for that.
For this case, I just happened to stumble over this because of a notification email and finally had the time and the opportunity to point it out.


> @gnn Your comment is quite crucial - in several tasks files are flying around in the task dir(s) but not in the working dir. A correct handling should be part of some best practice docs.

It already [is][0]. Unfortunately, due to my limited time and the low priority of things like this compared to other features, the document didn't get ready until development was already well underway and some facts, i.e. code, was already established. So I'm partially responsible for this. For other reasons too. (See below).

[0]: https://github.com/openego/eGon-data/blob/dev/CONTRIBUTING.rst#where-to-save-downloaded-data

> I think we won't fix this now but should definitely approach as soon as all tasks will have been migrated to datasets. I've opened an issue: #341

Again, such an issue [already exists][1]. There's even [an associated pull request][2] already, but it might be severely outdated. The original plan was that my new colleague takes care of this, but getting into development proved, and still proves, to take longer than expected. And I didn't put this issue back on the agenda, once it was foreseeable that my colleague would take longer for this than expected.

[1]: https://github.com/openego/eGon-data/issues/203
[2]: https://github.com/openego/eGon-data/pull/208